src/main/java/com/mathworks/ci/MatlabCommandStepExecution.java

```diff
@@ -59,7 +58,7 @@ public void stop(Throwable cause) throws Exception {
         getContext().onFailure(cause);
     }
     
-    private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
+    private int execMatlabCommand(FilePath workspace, Launcher launcher,
```

Comment:
Why did this previously need to be `synchronized`?

Reply:
Right, none of these required it if we had not set environment in perform method.